src/Providers/Maps/Google/Helper/RouteRenderer.ts

```diff
@@ -0,0 +1,115 @@
+namespace Provider.Maps.Google.Helper {
+	export class RouteRenderer implements OSFramework.Maps.Interface.IDisposable {
+		private _isRouteRendered: boolean;
+		private _map: OSMap.IMapGoogle;
+		private _pathPolyline: google.maps.Polyline;
+		private _routeMarkers: google.maps.marker.AdvancedMarkerElement[];
+
+		constructor(map: OSMap.IMapGoogle) {
+			this._map = map;
+			this._isRouteRendered = false;
+			this._routeMarkers = [];
+		}
+
+		private _buildMarker(
+			position: google.maps.LatLng | google.maps.LatLngLiteral,
+			label: string
+		): google.maps.marker.AdvancedMarkerElement {
+			// The pin element is the colored teardrop shape.
+			const pin = new google.maps.marker.PinElement({
+				background: '#EA4335', // Red background
+				borderColor: '#D6352D', // Dark red border
+				glyphColor: '#FFFFFF', // White letter
+				scale: 1,
+			});
+
+			// Create the advanced marker.
+			const marker = new google.maps.marker.AdvancedMarkerElement({
+				position: {
+					lat: Helper.Conversions.GetCoordinateValue(position.lat),
+					lng: Helper.Conversions.GetCoordinateValue(position.lng),
+				},
+				map: this._map.provider,
+				content: pin.element,
+			});
+
+			pin.glyph = label;
+
+			return marker;
+		}
+
+		public dispose(): void {
+			this.removeRoute();
+			this._map = undefined;
+			this._routeMarkers = [];
+			this._pathPolyline = undefined;
```

Comment:
This is done on removeRoute, and can be removed from here.

Reply:
done